website/content/plugins/drivers/virt/task-config.mdx

```diff
@@ -0,0 +1,155 @@
+---
+layout: docs
+page_title: Task Config
+description: Configure the virt task driver job specification parameters.
+---
+
+# Task configuration
+
+@include 'virt-beta-callout.mdx'
+
+Task configuration for an virt task includes setting an `image` to provide the base VM image as
+well as other parameters required for bootstrapping a VM instance.
+
+- `image` - (string: required) - Path to an `.img` cloud image to base the VM disk on. The image should
+  be located in an allowed path. The cloud image needs to include cloud-init, otherwise certain
+  task features are not available. You may also downloaded via the [artifact block][].
+
+- `use_thin_copy` - (bool: `false`) - Make a thin copy of the image using QEMU, and use it as the
+  backing cloud image for the VM.
+
+- `hostname` - (string: `"nomad-<task-name>"`) - The hostname to assign to the VM. As this value is
+  used as a network host name, it must be a valid DNS label according to RFC 1123. Nomad generates the default
+  value using the Nomad task name.
+
+- `cmds` - (list(string): optional) - List of commands to execute on the VM once it is running.
+  These commands can provide you with a quick way to start a process on the newly created
+  VM. When used in conjunction with the template, this command list can be a powerful start up tool.
+
+- `default_user_password` - (string: `""`) - Initial password for the default user
+  on the newly created VM. When set, the user must update the password on first connect.
+
+- `default_user_authorized_ssh_key` - (string: `""`) - SSH public key  added to the SSH
+  configuration for the default user of the cloud image distribution.
+
+- `user_data ` - (string: `""`) - Path to a cloud-init compliant user data file used as the user-data for the cloud-init configuration.
+
+- `primary_disk_size` - (int: required) - Disk space to assign to the VM. Must fit the VM's OS.
+
+Example:
+```hcl
+config {
+  image                           = "local/focal-server-cloudimg-amd64.img"
+  use_thing_copy                  = false
+  hostname                        = ""
+  cmds                            = ["python -m http.server 8000"]
+  default_user_password           = "password"
```

Comment:
Security scanners are going to ping us on this one. Mind commenting it out and/or making it a more obvious "do not actually use this" value?

```suggestion
  #default_user_password           = "CHANGE ME"
```